modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C

```diff
@@ -601,6 +645,170 @@ INSFVRhieChowInterpolator::getVelocity(const Moose::FV::InterpMethod m,
+          elem_value += (*this->_volumetric_force[i])(loc_boundary_face, time) *
+                        fi_loc->faceArea() *
+                        (fi_loc->faceCentroid() - elem->vertex_average()).norm() *
+                        (fi_loc->normal() * loc_normal);
+      }
+      else
+      {
+        Moose::FaceArg loc_face{
+            fi_loc, Moose::FV::LimiterType::CentralDifference, true, correct_skewness, nullptr};
+
+        for (const auto i : make_range(_volumetric_force.size()))
+          elem_value += (*this->_volumetric_force[i])(loc_face, time) * fi_loc->faceArea() *
+                        (fi_loc->faceCentroid() - elem->vertex_average()).norm() *
+                        (fi_loc->normal() * loc_normal);
+      }
+    }
+    elem_value /= elem->volume();
+
+    // Uncorrected interpolation - Step 2: loop over the face of the neighbor to compute
+    // face-average cell value
+    for (const auto side : make_range(neighbor->n_sides()))
+    {
+      const Elem * const loc_elem = neighbor->neighbor_ptr(side);
+      const bool elem_has_fi = Moose::FV::elemHasFaceInfo(*neighbor, loc_elem);
+      const FaceInfo * const fi_loc =
+          _moose_mesh.faceInfo(elem_has_fi ? neighbor : loc_elem,
+                               elem_has_fi ? side : loc_elem->which_neighbor_am_i(neighbor));
+
+      if (Moose::FV::onBoundary(*this, *fi_loc))
+      {
+        const Elem * const boundary_elem =
+            hasBlocks(neighbor->subdomain_id()) ? neighbor : loc_elem;
+        const Moose::FaceArg loc_boundary_face{fi_loc,
+                                               Moose::FV::LimiterType::CentralDifference,
+                                               true,
+                                               correct_skewness,
+                                               boundary_elem};
+
+        for (const auto i : make_range(_volumetric_force.size()))
+          neigh_value += (*this->_volumetric_force[i])(loc_boundary_face, time) *
+                         fi_loc->faceArea() *
+                         (fi_loc->faceCentroid() - neighbor->vertex_average()).norm() *
+                         (fi_loc->normal() * loc_normal);
+      }
+      else
+      {
+        Moose::FaceArg loc_face{
+            fi_loc, Moose::FV::LimiterType::CentralDifference, true, correct_skewness, nullptr};
+
+        for (const auto i : make_range(_volumetric_force.size()))
+          neigh_value += (*this->_volumetric_force[i])(loc_face, time) * fi_loc->faceArea() *
+                         (fi_loc->faceCentroid() - neighbor->vertex_average()).norm() *
+                         (fi_loc->normal() * loc_normal);
+      }
+    }
+    neigh_value /= neighbor->volume();
+
+    // Uncorrected interpolation - Step 3: interpolate element and neighbor reconstructed values
+    // to the face
+    interpolate(
+        Moose::FV::InterpMethod::Average, uncorrected_interp_vf, elem_value, neigh_value, fi, true);
+    uncorrected_interp_vf *= fi.gC();
```

Comment:
Yes. It is to make the two formulations consistent. I tried taking it out, but this yield back the pump oscillations

Reply:
It feels asymmetric. If one element centroid is closer to the face than the other, then the result of this calculation would change depending on the element numbering of the mesh (since we associate gC() with the lower ID element).

Reply:
It looks like this multiplication got removed which is cool. Are things still working as desired?

Reply:
Thanks. This line was removed.